app/assets/javascripts/discourse/tests/unit/lib/url-test.js

```diff
@@ -87,6 +87,30 @@ module("Unit | Utility | url", function (hooks) {
     assert.true(DiscourseURL.isInternal("#foo"), "anchor URLs are internal");
   });
 
+  test("isInternalTopic", function (assert) {
+    sinon.stub(DiscourseURL, "origin").get(() => "https://eviltrout.com");
+    assert.true(DiscourseURL.isInternalTopic("https://eviltrout.com/t/123"));
+    assert.false(DiscourseURL.isInternalTopic("https://eviltrout.com/admin"));
```

Comment:
I would add a test for an url such as `https://eviltrout.com/topic` (aka. starts with "/t" but not "/t/")

Reply:
Good call, done!